platforms/chibios/drivers/spi_master.c

```diff
@@ -283,9 +311,9 @@ spi_status_t spi_receive(uint8_t *data, uint16_t length) {
 }
 
 void spi_stop(void) {
-    if (currentSlavePin != NO_PIN) {
+    if (spiStarted) {
         spiUnselect(&SPI_DRIVER);
```

Comment:
```suggestion
#if SPI_SELECT_MODE == SPI_SELECT_MODE_NONE
        if (currentSlavePin != NO_PIN) {
            writePinHigh(currentSlavePin);
        }
#endif
        spiUnselect(&SPI_DRIVER);
```
(this is the only place where `currentSlavePin` is needed)